Tests/TracingTests/TracerTests.swift

```diff
@@ -63,11 +63,15 @@ final class TracerTests: XCTestCase {
             InstrumentationSystem.bootstrapInternal(NoOpTracer())
         }
 
+        var spanEnded = false
+        tracer.onEndSpan = { _ in spanEnded = true }
+
         let value = tracer.withSpan("hello", baggage: .topLevel) { _ in
             "yes"
         }
 
         XCTAssertEqual(value, "yes")
+        XCTAssertEqual(spanEnded, true)
```

Comment:
```suggestion
        XCTAssert(spanEnded)
```

Reply:
also `XCTAssertTrue` if you want to be more explicit

Reply:
```suggestion
        XCTAssertTrue(spanEnded)
```